sky/clouds/aws.py

```diff
@@ -160,7 +161,15 @@ def zones_provision_loop(
                                             region=region,
                                             zone=None)
         for r in regions:
-            yield r.zones
+            assert r.zones is not None, r
+            if num_nodes > 1:
+                # When num_nodes > 1, we need to try the zones one by
+                # one, to avoid the nodes of a same cluster being
+                # placed in different zones.
```

Comment:
```suggestion
                # When num_nodes > 1, we shouldn't pass a list of zones to the AWS NodeProvider to try, because it may then place the nodes of the same cluster in different zones. This is an artifact of the current AWS NodeProvider implementation.
```
(To be more clear, since "try the zones one by one" is technically what the node provider does)

Reply:
Good point! Fixed. Thanks!